client/src/pages/RegistrationPage.jsx

```diff
@@ -0,0 +1,221 @@
+import React, { useState } from 'react';
+import { Link } from 'react-router-dom';
+import { Formik, Form, Field } from 'formik';
+import cn from 'classnames';
+
+import { authService } from '../services/authService.js';
+import { usePageError } from '../hooks/usePageError.js';
+
+function validateName(value) {
+  if (!value) {
+    return 'Name is required';
+  }
+
+  if (value.includes(' ')) {
+    return 'One word required';
+  }
+
+  if (value.length < 2) {
+    return 'At least 2 characters';
+  }
+}
+
+function validateEmail(value) {
+  if (!value) {
+    return 'Email is required';
+  }
+
+  const emailPattern = /^[\w.+-]+@([\w-]+\.){1,3}[\w-]{2,}$/;
+
+  if (!emailPattern.test(value)) {
+    return 'Email is not valid';
+  }
+}
+
+const validatePassword = (value) => {
+  if (!value) {
+    return 'Password is required';
+  }
+
+  if (value.length < 6) {
+    return 'At least 6 characters';
+  }
+};
+
+export const RegistrationPage = () => {
+  const [error, setError] = usePageError('');
```

Comment:
The `usePageError` hook is being used incorrectly. It seems like it should return a function to set the error, but here it's being destructured as if it returns an array. Make sure to check the implementation of `usePageError` and use it correctly.